src/metrics.rs

```diff
@@ -85,64 +89,137 @@ pub fn metrics_to_string() -> std::result::Result<String, DshError> {
 /// #[tokio::main]
 /// async fn main() {
 ///    tokio::spawn(async move {
-///       start_http_server(8080).await;
+///       start_http_server(8080).await.unwrap();
 ///   });
 /// }
 /// ```
-pub async fn start_http_server(port: u16) {
-    let metrics_route = warp::path!("metrics").and_then(http_metric_response);
-    warp::serve(metrics_route).run(([0, 0, 0, 0], port)).await;
+///
+pub async fn start_http_server(port: u16) -> Result<()> {
+    let addr: SocketAddr = ([0, 0, 0, 0], port).into();
+    let listener = TcpListener::bind(addr).await?;
```

Comment:
I think a providing an error log is sufficient instead of returning a result. 
If the listeners fail for some reason, it does mean that the HTTP server will not start, but i think it's not a big issue.

Similar how you handled serve_connection error

Reply:
Skipping the Result in start_http_server fn doesn't make sense to me because we hide the error from the user. I think we need to propagate it and pass the handle error to the user

Reply:
I like the idea of giving the tokio join_handle that has a Result back to the user. Then the user is able to ignore the failure or handle it.